python/ray/tests/test_basic_5.py

```diff
@@ -124,34 +124,37 @@ def test_internal_kv(ray_start_regular):
         kv._internal_kv_list("@namespace_abc", namespace="n")
 
 
-def test_run_on_all_workers(ray_start_regular):
+def test_run_on_all_workers(ray_start_regular, tmp_path):
     # This test is to ensure run_function_on_all_workers are executed
     # on all workers.
-    @ray.remote
-    class Actor:
-        def __init__(self):
-            self.jobs = []
-
-        def record(self, job_id=None):
-            if job_id is not None:
-                self.jobs.append(job_id)
-            return self.jobs
-
-    a = Actor.options(name="recorder", namespace="n").remote()  # noqa: F841
-    driver_script = """
+    lock_file = tmp_path / "lock"
+    data_file = tmp_path / "data"
+    driver_script = f"""
 import ray
+from filelock import FileLock
 from pathlib import Path
+import pickle
+
+lock_file = "{str(lock_file)}"
+data_file = Path("{str(data_file)}")
 
 def init_func(worker_info):
-    a = ray.get_actor("recorder", namespace="n")
-    a.record.remote(worker_info['worker'].worker_id)
```

Comment:
The old code has a race condition. Before the worker calls run in loop, we can't call any remote function. 

If init_func is called before the worker calls run in loop, it'll crash.
If init_func is called after the worker executes tasks, it'll be wrong.